pyanaconda/modules/boss/kickstart_manager.py

```diff
@@ -0,0 +1,139 @@
+
+    def __init__(self):
+        self._kickstart_path = None
+        self._elements = None
+        self._module_observers = []
+
+    @property
+    def module_observers(self):
+        """Get all module observers for kickstart distribution."""
+
+        return self._module_observers
+
+    @module_observers.setter
+    def module_observers(self, modules):
+        """Set module observers for kicstart distribution.
+
+        :param modules: Module observers list
+        :type modules: list(DBusObjectObserver)
+        """
+        self._module_observers = modules
+
+    @property
+    def elements(self):
+        """Return all elements of split kickstart."""
+        return self._elements
+
+    @property
+    def unprocessed_kickstart(self):
+        """Return kickstart not processed by any module."""
+        return self._elements.get_kickstart_from_elements(self._elements.unprocessed_elements)
+
+    def split(self, path):
+        """Split the kickstart given by path into elements."""
+        self._elements = None
+        self._kickstart_path = path
+        handler = makeVersion()
+        ksparser = SplitKickstartParser(handler, valid_sections=VALID_SECTIONS_ANACONDA)
+        try:
+            result = ksparser.split(path)
+        except KickstartParseError as e:
+            raise SplitKickstartUnknownSectionError(e)
+        except KickstartError as e:
+            raise SplitKickstartMissingIncludeError(e)
+        log.info("split {}: {}".format(path, result))
+        self._elements = result
+
+    def distribute(self):
+        """Distribute split kickstart to modules synchronously.
+
+        :returns: list of (Line number, Message) errors reported by modules when
+                  distributing kickstart
+        :rtype: list((int, str))
+        """
+        errors = []
+
+        for observer in self._module_observers:
+
+            if not observer.is_service_available:
+                log.warning("distribute kickstart: module {} not available".format(observer.service_name))
+                continue
+
+            commands = observer.proxy.KickstartCommands()
+            sections = observer.proxy.KickstartSections()
+            addons = observer.proxy.KickstartAddons()
+            log.info("distribute kickstart: {} handles commands {} sections {} addons {}".format(
+                observer.service_name, commands, sections, addons))
+
+            elements = self._elements.get_and_process_elements(commands=commands,
+                                                               sections=sections,
+                                                               addons=addons)
+            kickstart = self._elements.get_kickstart_from_elements(elements)
+            log.info("distribute kickstart: {} will get kickstart elements: {}".format(
+                observer.service_name, elements))
+
+            error_lineno, error_msg = observer.proxy.ConfigureWithKickstart(kickstart)
```

Comment:
We might need to get the warnings are well, because in the strict mode we treat them as errors. But that can be extended in another issue.